integration/mediation-tests/tests-service/src/test/java/org/wso2/carbon/esb/car/deployment/test/ClassMediatorCarTestCase.java

```diff
@@ -20,58 +20,31 @@
 
 import org.apache.axiom.om.OMElement;
 import org.apache.axis2.AxisFault;
-import org.apache.commons.lang.ArrayUtils;
 import org.testng.Assert;
-import org.testng.annotations.AfterClass;
 import org.testng.annotations.BeforeClass;
 import org.testng.annotations.Test;
-import org.wso2.carbon.integration.common.admin.client.ApplicationAdminClient;
-import org.wso2.carbon.integration.common.admin.client.CarbonAppUploaderClient;
 import org.wso2.esb.integration.common.utils.ESBIntegrationTest;
-
-import java.io.File;
-import java.util.Calendar;
 import java.util.concurrent.TimeUnit;
-import javax.activation.DataHandler;
-import javax.activation.FileDataSource;
 
 public class ClassMediatorCarTestCase extends ESBIntegrationTest {
 
-    private CarbonAppUploaderClient carbonAppUploaderClient;
-    private ApplicationAdminClient applicationAdminClient;
-    private final int MAX_TIME = 120000;
-    private final String car1Name = "MediatorCApp_1.0.0";
-    private final String car2Name = "MediatorCApp2_1.0.0";
-    private final String car1FileName = car1Name + ".car";
-    private final String car2FileName = car2Name + ".car";
-    private final String proxyName = "MediatorTestProxy";
-    private boolean isCarFile1Uploaded = false;
-    private boolean isCarFile2Uploaded = false;
+    private final String car1Name = "MediatorCApp";
 
     @BeforeClass(alwaysRun = true, description = "Test Car with Mediator deployment")
     protected void uploadCar1Test() throws Exception {
         super.init();
-        carbonAppUploaderClient = new CarbonAppUploaderClient(context.getContextUrls().getBackEndUrl(),
-                getSessionCookie());
-        carbonAppUploaderClient.uploadCarbonAppArtifact(car1FileName, new DataHandler(new FileDataSource(
-                new File(getESBResourceLocation() + File.separator + "car" + File.separator + car1FileName))));
-        isCarFile1Uploaded = true;
-        applicationAdminClient = new ApplicationAdminClient(context.getContextUrls().getBackEndUrl(),
-                getSessionCookie());
-        Assert.assertTrue(isCarFileDeployed(car1Name), "Car file deployment failed");
+        Assert.assertTrue(checkCarbonAppExistence(car1Name), "Car file deployment failed");
         TimeUnit.SECONDS.sleep(5);
     }
 
     @Test(groups = { "wso2.esb" }, description = "Test Car with Mediator deployment and invocation")
     public void capp1DeploymentAndServiceInvocation() throws Exception {
 
-        Assert.assertTrue(
-                esbUtils.isProxyDeployed(context.getContextUrls().getBackEndUrl(), getSessionCookie(), proxyName),
-                "Proxy service deployment failed");
+        Assert.assertTrue(checkProxyServiceExistence("MediatorTestProxy"),"Proxy service deployment failed");
```

Comment:
lets remove this , since we are invoking the proxy it must fail at that point if it doesn't exist and by removing this check we can exclude a redundant call.

Reply:
Remove the proxy invoke call. Please refer the latest commit.